pkg/services/rpcsrv/client_test.go

```diff
@@ -1135,6 +1135,127 @@ func TestSignAndPushP2PNotaryRequest(t *testing.T) {
+
+	c, err := rpcclient.New(context.Background(), httpSrv.URL, rpcclient.Options{})
+	require.NoError(t, err)
+	t.Run("getrawnotarypool", func(t *testing.T) {
+		t.Run("empty pool", func(t *testing.T) {
+			np, err := c.GetRawNotaryPool()
+			require.NoError(t, err)
+			require.Equal(t, 0, len(np.Hashes))
+		})
+
+		sender := testchain.PrivateKeyByID(0) // owner of the deposit in testchain
+		acc := wallet.NewAccountFromPrivateKey(sender)
+
+		comm, err := c.GetCommittee()
+		require.NoError(t, err)
+
+		multiAcc := &wallet.Account{}
+		*multiAcc = *acc
+		require.NoError(t, multiAcc.ConvertMultisig(smartcontract.GetMajorityHonestNodeCount(len(comm)), comm))
+
+		nact, err := notary.NewActor(c, []actor.SignerAccount{{
+			Signer: transaction.Signer{
+				Account: multiAcc.Contract.ScriptHash(),
+				Scopes:  transaction.CalledByEntry,
+			},
+			Account: multiAcc,
+		}}, acc)
+		require.NoError(t, err)
+		neoW := neo.New(nact)
+		// Send the 1st notary request
+		tx1, err = neoW.SetRegisterPriceTransaction(1_0000_0000)
+		require.NoError(t, err)
+		mainHash1, fallbackHash1, _, err = nact.Notarize(tx1, err)
+		require.NoError(t, err)
+
+		checkTxInActPool := func(t *testing.T, mainHash, fallbackHash util.Uint256, actNotaryPool *result.RawNotaryPool) {
+			actFallbacks, ok := actNotaryPool.Hashes[mainHash]
+			require.Equal(t, true, ok)
+			require.Equal(t, 1, len(actFallbacks))
+			require.Equal(t, fallbackHash, actFallbacks[0])
+		}
+		t.Run("nonempty pool", func(t *testing.T) {
+			actNotaryPool, err := c.GetRawNotaryPool()
+			require.NoError(t, err)
+			require.Equal(t, 1, len(actNotaryPool.Hashes))
+			checkTxInActPool(t, mainHash1, fallbackHash1, actNotaryPool)
+		})
+
+		// Send the 2nd notary request
+		tx2, err = neoW.SetRegisterPriceTransaction(2_0000_0000)
+		require.NoError(t, err)
+		mainHash2, fallbackHash2, _, err = nact.Notarize(tx2, err)
+		require.NoError(t, err)
+
+		t.Run("pool with 2", func(t *testing.T) {
+			actNotaryPool, err := c.GetRawNotaryPool()
+			require.NoError(t, err)
+			require.Equal(t, 2, len(actNotaryPool.Hashes))
+			checkTxInActPool(t, mainHash1, fallbackHash1, actNotaryPool)
+			checkTxInActPool(t, mainHash2, fallbackHash2, actNotaryPool)
+		})
+	})
+	t.Run("getrawnotarytransaction", func(t *testing.T) {
+		compareTx := func(t *testing.T, expectedTx, actualTx *transaction.Transaction) func(t *testing.T) {
+			return func(t *testing.T) {
+				require.Equal(t, expectedTx.Hash(), actualTx.Hash())
+				require.Equal(t, expectedTx.Size(), actualTx.Size())
+			}
+		}
+		t.Run("client GetRawNotaryTransaction", func(t *testing.T) {
+			t.Run("unknown transaction", func(t *testing.T) {
+				_, err := c.GetRawNotaryTransaction(util.Uint256{0, 0, 0})
+				require.Error(t, err)
+				require.ErrorIs(t, err, neorpc.ErrUnknownTransaction)
+			})
+			t.Run("transactions from pool", func(t *testing.T) {
+				mainTx1, err := c.GetRawNotaryTransaction(mainHash1)
+				require.NoError(t, err)
+				compareTx(t, tx1, mainTx1)
```

Comment:
And just FYI, there's no call to comparing function. `compareTx` returns a function, and you need to call it: `compareTx(t, tx1, mainTx1)(t)`, otherwise the test will pass even with wrong `mainTx1`.